packages/shared/src/hooks/useFeed.ts

```diff
@@ -179,56 +181,110 @@ export default function useFeed<T>(
     refetchOnReconnect: false,
     refetchOnWindowFocus: false,
   });
+  const {
+    data: adsData,
+    fetchNextPage: fetchNextAd,
+    isLoading,
+    dataUpdatedAt: adsUpdatedAt,
+  } = adsQuery;
 
-  useEffect(() => {
-    if (
-      !adsQuery.isFetching &&
-      adsQuery.data?.pages?.length < feedQuery.data?.pages?.length
-    ) {
-      adsQuery.fetchNextPage();
-    }
-    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [adsQuery.data, feedQuery.data, adsQuery.isFetching]);
+  const getAd = useCallback(
+    ({ index }: { index: number }) => {
+      if (!isAdsQueryEnabled) {
+        return undefined;
+      }
+
+      const adStart =
+        adTemplate.adStart ??
+        featureFeedAdTemplate.defaultValue.default.adStart;
```

Comment:
Do you need this one here still?
I think you do the swap in FeedContext already right? (to retro fix on empty)

Reply:
just in case its not defined for some reason to prevent everything from breaking

Reply:
But the setting relies on the same thing right?
Could it even be empty in 1 not the other then?

Reply:
`adSpot` is required, so it should not be empty in any of the defined ones

Reply:
this is if all else fails scenario

Reply:
Yeah rather safe I guess, let's leave it, doesn't make much difference.